gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleProjectProvider.java

```diff
@@ -82,52 +95,181 @@ public synchronized ILiferayProject provide( Object adaptable )
     @Override
     public IStatus createNewProject( NewLiferayModuleProjectOp op, IProgressMonitor monitor ) throws CoreException
     {
-        IStatus retval = null;
+        IStatus retval = Status.OK_STATUS;
 
         final String projectName = op.getProjectName().content();
 
         IPath location = PathBridge.create( op.getLocation().content() );
 
+        String className = op.getComponentName().content();
+
+        final String serviceName = op.getServiceName().content();
+
+        final String packageName = op.getPackageName().content();
+
+        ElementList<PropertyKey> propertyKeys = op.getPropertyKeys();
+
+        final List<String> properties = new ArrayList<String>();
+
+        for( PropertyKey propertyKey : propertyKeys )
+        {
+            properties.add( propertyKey.getName().content( true ) + "=" + propertyKey.getValue().content( true ) );
+        }
+
         final String projectTemplateName = op.getProjectTemplateName().content();
 
         StringBuilder sb = new StringBuilder();
         sb.append( "create " );
-        sb.append( "-d \"" + location.toFile().getAbsolutePath() +  "\" " );
+        sb.append( "-d \"" + location.toFile().getAbsolutePath() + "\" " );
         sb.append( "-t " + projectTemplateName + " " );
-        sb.append( "\"" + projectName + "\"");
 
-        try
+        if( className != null )
         {
-            BladeCLI.execute( sb.toString() );
+            sb.append( "-c " + className + " " );
         }
-        catch( BladeCLIException e )
+
+        if( serviceName != null )
         {
-            retval = ProjectCore.createErrorStatus( e );
+            sb.append( "-s " + serviceName + " " );
         }
 
-        if( retval.isOK() )
+        if( packageName != null )
+        {
+            sb.append( "-p " + packageName + " " );
+        }
+
+        sb.append( "\"" + projectName + "\" " );
+
+        try
+        {
+            final String[] ret = BladeCLI.execute( sb.toString() );
+
+            final String errors = checkForErrors(ret);
+
+            if( errors.length() > 0 )
+            {
+                retval = GradleCore.createErrorStatus( "Project create error: " + errors );
+                return retval;
+            }
+
+            if( ret.length == 0 )
+            {
+                ElementList<ProjectName> projectNames = op.getProjectNames();
+                
+                projectNames.insert().setName( projectName );
+                
+                IPath projecLocation = location;
+
+                final String lastSegment = location.lastSegment();
+
+                if( location != null && location.segmentCount() > 0 )
+                {
+                    if( !lastSegment.equals( projectName ) )
+                    {
+                        projecLocation = location.append( projectName );
+                    }
+                }
+
+                ProjectImportConfiguration configuration = new ProjectImportConfiguration();
```

Comment:
Lets find the java file to replace the properties first, and then import the gradle project.